### PR TITLE
OKTA-408478: override contentLength() method in InputStreamRequestBody

### PR DESCRIPTION
**Pull request by VitaliiTytarenko-okta**
Issue https://github.com/okta/okta-commons-java/issues/52

### CODE REVIEW

File: http/okhttp/src/main/java/com/okta/commons/http/okhttp/OkHttpRequestExecutor.java [hunk 1]
```diff
@@ -196,5 +196,10 @@ public void writeTo(BufferedSink sink) throws IOException {
                 Util.closeQuietly(source);
             }
         }
+
+        @Override
+        public long contentLength() {
+            return ((ByteArrayInputStream) inputStream).available();
```
Comment (bdemers) [inline http/okhttp/src/main/java/com/okta/commons/http/okhttp/OkHttpRequestExecutor.java:202]: Avoid casting here, if this implementation requires a BAIS, then set the type to match:
```diff
- private final InputStream inputStream;
+ private final ByteArrayInputStream inputStream;
```

Comment (bdemers) [reply http/okhttp/src/main/java/com/okta/commons/http/okhttp/OkHttpRequestExecutor.java:202]: Does this need a null check?

**Review by bdemers — COMMENTED** [review]
A couple of nits below.

Also, I'm not against this change, but if an unset `Content-Length` header is causing a problem we need to fix that on the other end.

File: http/okhttp/src/main/java/com/okta/commons/http/okhttp/OkHttpRequestExecutor.java (hunk 1 above)
Comment (VitaliiTytarenko-okta) [reply http/okhttp/src/main/java/com/okta/commons/http/okhttp/OkHttpRequestExecutor.java]: Yeah, sorry for this.
Fixed.

**Comment by arvindkrishnakumar-okta** [comment]
> A couple of nits below.
> 
> Also, I'm not against this change, but if an unset `Content-Length` header is causing a problem we need to fix that on the other end.

FWIW, Okta documentation states that a missing CL header will result in HTTP 411 error - https://developer.okta.com/docs/reference/api-overview/#delete

**Comment by arvindkrishnakumar-okta** [comment]
> > A couple of nits below.
> > Also, I'm not against this change, but if an unset `Content-Length` header is causing a problem we need to fix that on the other end.
> 
> FWIW, Okta documentation states that a missing CL header will result in HTTP 411 error - https://developer.okta.com/docs/reference/api-overview/#delete

Apparently, this seems to be a documentation issue and needs to be updated. Flagged this with internal doc team.

**Review by arvindkrishnakumar-okta — APPROVED** [review]

**Pull request merged**